File: leetcode/28-find-the-Index-of-the-first-occurrence-in-a-string.ts
function strStr(haystack: string, needle: string): number {
    // Пустая строка всегда найдена в начале
    if (!needle) return 0;

    const m = needle.length;

    for (let i = 0; i <= haystack.length - m; i++) {
        let j = 0;

        // Проверяем посимвольно
        while (j < m && haystack[i + j] === needle[j]) j++;

        // Если дошли до конца needle, значит нашли вхождение
        if (j === m) return i;
    }

    return -1;
}

strStr("hello", "ll"); // 2

File: go/leetcode/28-find-the-Index-of-the-first-occurrence-in-a-string.go
package main

func strStr(haystack string, needle string) int {
	m := len(needle)

	if m == 0 {
		return 0
	}

	for i := 0; i <= len(haystack) - m; i++ {
		j := 0

		for j < m {
			if haystack[i+j] != needle[j] {
				break
			}
			j++
		}

		if j == m {
			return i
		}
	}

	return -1
}
